### PR TITLE
Xtheme: add feature to add global snippet injection

### PR DESCRIPTION
**Pull request by chessbr**
Add feature to inject snippet of codes in all or a set of themes.

Add widget to enable CoreMirror editor

Refs POS-2564

### CODE REVIEW

File: shuup/xtheme/models.py [hunk 1]
```diff
@@ -160,3 +173,22 @@ def update_settings(self, update_values):
...
+class Snippet(models.Model):
+    """
+    Inject snippet code globally filtering by themes if configured
+    """
+    shop = models.ForeignKey("shuup.Shop", related_name="snippets")
+    location = models.CharField(max_length=64, verbose_name=_("location"))
+    snippet_type = models.CharField(max_length=20, verbose_name=_("snippet type"), choices=SnippetTypeChoices)
+    snippet = models.TextField(verbose_name=_("snippet"))
+    # list of theme identifiers that will be have this sniipet injected, if None, it means all themes
+    themes = SeparatedValuesField(verbose_name=_("themes"), blank=True, null=True)
```
Comment (tulimaki) [inline shuup/xtheme/models.py]: Why we need themes here?

File: shuup/xtheme/resources.py [hunk 2]
```diff
@@ -213,3 +250,49 @@ def add_resource(context, location, resource):
...
+
+def inject_global_snippet(context, content):
+    if not valid_view(context):
+        return
+
+    from shuup.xtheme import get_current_theme
+    from shuup.xtheme.models import Snippet, SnippetType
+    shop = get_shop(context["request"])
+
+    for snippet in Snippet.objects.filter(shop=shop):
```
Comment (tulimaki) [inline shuup/xtheme/resources.py]: This likely needs to be cached. We can't loop over snippets on every page load.

File: shuup/xtheme/admin_module/__init__.py [hunk 3]
```diff
@@ -96,3 +98,45 @@ def get_notifications(self, request):
                         title=_("Theming"),
                         url="shuup_admin:xtheme.config"
                     )
+
+
+class XthemeSnippetsAdminModule(AdminModule):
```
Comment (tulimaki) [inline shuup/xtheme/admin_module/__init__.py]: Make this available from theme configuration (action button over table) and the front toolbar.

File: shuup/xtheme/models.py (hunk 1 above)
Comment (chessbr) [reply shuup/xtheme/models.py]: you can enable injection just for some themes

File: shuup/xtheme/resources.py (hunk 2 above)
Comment (chessbr) [reply shuup/xtheme/resources.py]: true

File: shuup/xtheme/models.py (hunk 1 above)
Comment (tulimaki) [reply shuup/xtheme/models.py]: Make sure optional then and add help text too.

Comment (chessbr) [reply shuup/xtheme/models.py]: it is optional, I just added the help text to the form, I will move it to here

**Pull request merged**